Remotion/ObjectBinding/Web/UI/Controls/BocListImplementation/Rendering/BocColumnRendererArrayBuilder.cs

```diff
@@ -63,14 +64,32 @@ public BocColumnRenderer[] CreateColumnRenderers ()
 
       PrepareSorting(sortingDirections, sortingOrder);
 
-      var firstValueColumnRendered = false;
+      var renderedRowIcon = false;
       var bocColumnRenderers = new List<BocColumnRenderer>(_columnDefinitions.Count);
       var visibleColumnIndex = GetInitialVisibleColumnIndex();
+
+      var enableIconOnFirstPossibleColumn = _columnDefinitions.All(
+          cd =>
+          {
+            return cd is not IBocColumnDefinitionWithRowIconSupport { RowIconEnabled: RowIcon.Preferred };
+          });
```

Comment:
It would be better to use Any to test for preferred and then use a `not` at the "correct" spot. That way, you only need to iterate until you find the first Preferred column, not check all the columns.